packages/connector/src/provider.ts

```diff
@@ -109,8 +132,18 @@ export type GianoProvider = EIP1193Provider & {
   request: EIP1193RequestFn<GianoProviderCustomMethods>
 }
 
-export const createGianoProvider = ({ transports, chains, initialChainId, bundler, injection, gianoSmartWalletFactoryAddress }: CreateGianoProviderParams) => {
-  let smartAccount: SmartAccount<GianoSmartAccountImplementation> | null;
+export let smartAccount: SmartAccount<GianoSmartAccountImplementation> | null = null;
```

Comment:
I think we should keep `smartAccount` scoped to `createGianoProvider` function, it is not a good idea to export a mutable global variable. This can also cause problems if we create multiple providers